### PR TITLE
[MCC-1312078] Fix/ ports and warnings

### PR DESCRIPTION
**Pull request by csavage-mdsol**
This resolves issue #4  and #2, thereby making the test service 10% more awesome.   5% to go!

### CODE REVIEW

File: lib/crichton_test_service.rb [hunk 1]
```diff
@@ -27,7 +29,17 @@ def self.wait_for_response!(url)
       connection.get('/')
     rescue Faraday::Error::ConnectionFailed
       sleep 1
-      retry if (i += 1)  < 30
+      retry if (i += 1)  <= 30
```
Comment (HonoreDB) [inline lib/crichton_test_service.rb]: Probably want `(i+=1) <= 30 ? retry : raise`

Comment (csavage-mdsol) [inline lib/crichton_test_service.rb]: Ah, yes.  I also want an ensure pid above.  Oops.

Comment (csavage-mdsol) [inline lib/crichton_test_service.rb]: I'm arguing with myself about how to test this behavior, btw.  Anything jump to mind?

Comment (HonoreDB) [inline lib/crichton_test_service.rb]: Bleh, seems like you'd need to stub sleep on the module, and then you're just mocking the thing you're testing.

Comment (csavage-mdsol) [inline lib/crichton_test_service.rb]: eewwwww

**Comment by csavage-mdsol** [comment]
```
Finished in 11.93 seconds (files took 0.85788 seconds to load)
9 examples, 0 failures, 6 pending

Randomized with seed 26175
```

**Comment by yzhangmedidata** [comment]
Looks good, merging in 5 mins

**Pull request merged**